src/js/common/validator.js

```diff
@@ -1,122 +1,470 @@
+                error.message = gpii.schema.validator.errorHintForRule(schemaPath, gssSchema, ajvError.message);
+
+                return error;
+            });
+            return {
+                isValid: false,
+                errors: transformedErrors
+            };
+        }
+        else {
+            return { isValid: true, errors: [] };
+        }
+    };
+
+    fluid.registerNamespace("gpii.schema.validator");
+    gpii.schema.validator.defaultLocalisationTransformRules = {
+        "data": "data",
+        "error": "error"
+    };
+
+    /**
+     *
+     * A function to translate/localise validation errors.
+     *
+     * If you want to pass a custom message bundle to this function, it should only contain top-level elements, see
+     * ./src/js/validation-errors.js in this package for an example.
+     *
+     * @param {Array<Object>} validationErrors - An array of validation errors, see `gpii.schema.validator.standardiseAjvErrors` for details.
+     * @param {Any} validatedData - The (optional) data that was validated.
+     * @param {Object} messages - An (optional) map of message templates (see above).  Defaults to the message bundle provided by this package.
+     * @param {Object} localisationTransform - An optional set of rules that control what information is available when localising validation errors (see above).
+     * @return {Array<Object>} - The validation errors, with all message keys replaced with localised strings.
+     *
+     */
+    gpii.schema.validator.localiseErrors = function (validationErrors, validatedData, messages, localisationTransform) {
+        messages = messages || gpii.schema.messages.validationErrors;
+        localisationTransform = localisationTransform || gpii.schema.validator.defaultLocalisationTransformRules;
+        var localisedErrors = fluid.transform(validationErrors, function (validationError) {
+            var messageKey = fluid.get(validationError, "message");
+            var messageTemplate = messageKey && fluid.get(messages, messageKey);
+            if (messageTemplate) {
+                var data = validatedData && fluid.get(validatedData, validationError.dataPath);
+                var localisationContext = fluid.model.transformWithRules({ data: data, error: validationError}, localisationTransform);
+                var localisedMessage = fluid.stringTemplate(messageTemplate, localisationContext);
+                return fluid.merge({}, validationError, { message: localisedMessage});
+            }
+            else {
+                return validationError;
+            }
+        });
+        return localisedErrors;
+    };
+
+    /**
+     *
+     * Convert a GSS schema to a standard JSON Schema.
+     *
+     * NOTE: This function is a non-API function, i.e. one that assists public functions in performing their work, but
+     * which is not guaranteed to remain available.
+     *
+     * @param {Object} originalFsl - The FSL schema.
```

Comment:
Presumably FSL here needs to be renamed GSS

Reply:
I searched for and replaced all variations throughout the project, including this one.